dash/_utils.py

```diff
@@ -108,11 +120,48 @@ def first(self, *names):
 
 def create_callback_id(output):
     if isinstance(output, (list, tuple)):
-        return '..{}..'.format('...'.join(
-            '{}.{}'.format(x.component_id, x.component_property)
-            for x in output
-        ))
+        return "..{}..".format(
+            "...".join(
+                "{}.{}".format(x.component_id, x.component_property)
+                for x in output
+            )
+        )
 
-    return '{}.{}'.format(
-        output.component_id, output.component_property
-    )
+    return "{}.{}".format(output.component_id, output.component_property)
+
+
+def run_command_with_process(cmd):
+    proc = subprocess.Popen(shlex.split(cmd, posix=sys.platform != "win32"))
+    proc.wait()
+    if proc.poll() is None:
+        logger.warning(u"🚨 trying to terminate subprocess in safe way")
+        try:
+            proc.communicate()
+        except Exception:  # pylint: disable=broad-except
+            logger.exception(u"🚨 first try communicate failed")
+            proc.kill()
+            proc.communicate()
+
+
+def compute_md5(path):
+    with open(path) as fp:
+        content = fp.read()
+        try:
+            buf = unicode(content, 'utf-8').encode('utf-8')  # noqa:F821
```

Comment:
always a headache for unicode with python2 and 3

Reply:
Ha yeah, I suppose we could use `open(path, 'b')` to get consistent bytes, but then we need something like `fp.read().replace(b'\r\n', b'\n')` to clean up windows line ending inconsistencies... would be shorter code but I can't say it's any clearer.

Reply:
I think we should seriously consider using the `future` module and its [idioms](https://python-future.org/compatible_idioms.html).

Reply:
Nice, I hadn't seen that collection of idioms 💯 
And yes, I forgot about the `encoding` kwarg - that's the right way to do it 🏆